homeassistant/components/ozw/light.py

```diff
@@ -118,6 +210,35 @@ async def async_turn_on(self, **kwargs):
         """Turn the device on."""
         self.async_set_duration(**kwargs)
 
+        rgbw = None
+
+        if ATTR_WHITE_VALUE in kwargs:
+            self._white = kwargs[ATTR_WHITE_VALUE]
```

Comment:
The entity state holding instance attributes should be updated by the node values by the device and not in the service call.

Reply:
So in `on_value_update` then if I fully understand correct?

Reply:
Moving these out of the service call broke the color calculations.

Reply:
I think I understand what to do after looking at the code for a few.

Reply:
Resolved in latest commit